pat/consumer_tests/step_6.py

```diff
@@ -0,0 +1,469 @@
+                operation.Handle,
+                operation_timeout,
+                extra={'step': step},
+            )
+            test_results.append(False)
+            continue
+        if len(operation_result.report_parts) != len(expected_order):
+            logger.error(
+                'The SetString operation with the handle %s finished with %d report parts, but expected exactly %d',
+                operation.Handle,
+                len(operation_result.report_parts),
+                len(expected_order),
+                extra={'step': step},
+            )
+            test_results.append(False)
+            continue
+        for report in (
+            report
+            for report in operation_result.report_parts
+            if report.InvocationInfo.InvocationState not in expected_order
+        ):
+            logger.error(
+                'The SetString operation with the handle %s returned unexpected invocation state %s with error %s: %s',
+                operation.Handle,
+                report.InvocationInfo.InvocationState,
+                report.InvocationInfo.InvocationError,
+                report.InvocationInfo.InvocationErrorMessage,
+                extra={'step': step},
+            )
+            test_results.append(False)
+        for expected, actual in zip(
+            expected_order,
+            (p.InvocationInfo.InvocationState for p in operation_result.report_parts),
+            strict=True,
+        ):
+            if expected == actual:
+                logger.info(
+                    'The answer of the SetString operation with the handle %s operation is "%s" as expected',
+                    operation.Handle,
+                    expected,
+                    extra={'step': step},
+                )
+            else:
+                logger.error(
+                    'The answer of the SetString operation with the handle %s operation is "%s", but expected "%s"',
+                    operation.Handle,
+                    actual,
+                    expected,
+                    extra={'step': step},
+                )
+                test_results.append(False)
+        if not any(
+            expected == actual
+            for expected, actual in zip(
+                expected_order,
+                (p.InvocationInfo.InvocationState for p in operation_result.report_parts),
+                strict=True,
+            )
+        ):
+            test_results.append(False)
+        else:
+            test_results.append(True)
+    return any(test_results) and all(test_results)
+
+
+def test_6e(consumer: SdcConsumer) -> bool:  # noqa: C901, PLR0912
+    """The Reference Consumer invokes SetMetricStates:
+    - Payload: 2 metric states (settings; consider alert limits)
+    - The Reference Provider immediately responds with Fin
+    - Action: The Reference Provider alters values of given metrics
```

Comment:
what do you think - is it needed to test this?

Reply:
So now you are testing that the provided value is the same as the received value. But here it states that it shall be "altered" / changed. So I understand the requirement here so that the value is not the one which was proposed during the operation invocation. So the current implementation will fail with other providers. And even the sdc11073 pat provider must change the proposed value to another one. What do you think about this?

Reply:
https://github.com/ornet-ev/plug-a-thon-testing/issues/24